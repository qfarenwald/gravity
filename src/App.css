.App {
  align-items: center;
  background-color: #000000;
  display: flex;
  flex-direction: column;
  /* font-family: 'Open Sans Condensed', sans-serif; */
  height: 100vh;
  justify-content: center;
  text-align: center;
  width: 100vw;
}
