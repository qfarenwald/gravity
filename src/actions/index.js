export const getTrails = (trails) => ({
  type: 'GET_TRAILS',
  trails,
});

export const makeBooking = (booking) => ({
  type: 'MAKE_BOOKING',
  booking,
});
