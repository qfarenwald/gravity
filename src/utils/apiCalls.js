export const fetchData = async (url) => {
  const response = await fetch(url);
  if (response.ok) {
    const data = await response.json();
    return data;
  }
  throw Error(response.statusText);
};
