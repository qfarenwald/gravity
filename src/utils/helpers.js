// clean to filter through and find trails with a decent greater than ?

export const filteredTrailData = trails => {
  return trails.map(trail => ({
    id: trail.id,
    name: trail.name,
    stars: trail.stars,
    summary: trail.summary,
    decent: trail.decent,
    difficulty: trail.difficulty,
    conditonDate: trail.conditonDate,
    conditionStatus: trail.conditionStatus,
    latitude: trail.latitude,
    longitude: trail.longitude
  }))
}
