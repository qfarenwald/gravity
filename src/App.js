import React, { Component } from 'react';
import { fetchData } from './utils/apiCalls';
import { filteredTrailData } from './utils/helpers';
import { connect } from 'react-redux';
import { getTrails } from './actions';
import { bindActionCreators } from 'redux';
import { Route } from 'react-router-dom';
import { Header } from './components/Header/Header';
import { Location } from './components/Location/Location';
import ShuttleForm from './containers/ShuttleForm/ShuttleForm';
import TrailsContainer from './containers/TrailsContainer/TrailsContainer';
import './App.css';

export class App extends Component {

// if doing multiple locations, take out of componentDidMount and on a button click to move from home page to trails page
// if ^ interpolate fetch with diff coordinates?
// move App to either components or container

  async componentDidMount() {
    console.log('app props', this.props)
    const { getTrails } = this.props
    console.log('getTrails', getTrails)

    try {
      const trails = await fetchData('https://www.mtbproject.com/data/get-trails?lat=39.7392&lon=-104.9903&maxDistance=10&key=200628346-0f130fc8870531d529e09b85e721317a')
      const cleanTrails = filteredTrailData(trails.trails)
      getTrails(cleanTrails)
    } catch {
      console.log('error')
    }
  }

  render() {
    return (
      <div className="App">
        <Route path='/' render={() => <Header />} />
        <Route exact path='/' render={() => <Location />} />
        <Route exact path='/trails' render={() => <TrailsContainer />} />
        <Route exact path='/shuttle' render={() => <ShuttleForm />} />
      </div>
    )
  }
}

export const mapDispatchToProps = dispatch => (
  bindActionCreators(
    {
      getTrails
    },
  dispatch)
)

export default connect(null, mapDispatchToProps)(App);
