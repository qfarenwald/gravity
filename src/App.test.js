import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import { App, mapDispatchToProps } from './App';
import { fetchData } from './utils/apiCalls';
import { getTrails } from './actions';
import { filteredTrailData } from './utils/helpers';

jest.mock('./utils/apiCalls');

describe('App', () => {
  let wrapper;

  const mockGetTrails = jest.fn();

  const mockResponse = {
    trails: [
      {
        key: 46286,
        id: 46286,
        name: 'Betasso Preserve',
        stars: 4,
        difficulty: 'greenBlue',
        descent: -829,
        conditionDate: '2019-10-03 22:57:25',
        conditionStatus: 'All Clear',
      },
      {
        key: 601365,
        id: 601365,
        name: 'Walker Ranch',
        stars: 3.9,
        difficulty: 'blueBlack',
        descent: -1510,
        conditionDate: '2019-09-19 07:48:20',
        conditionStatus: 'All Clear',
      },
    ],
  };

  const mockTrails = [
    {
      key: 46286,
      id: 46286,
      name: 'Betasso Preserve',
      stars: 4,
      difficulty: 'greenBlue',
      descent: -829,
      conditionDate: '2019-10-03 22:57:25',
      conditionStatus: 'All Clear',
    },
    {
      key: 601365,
      id: 601365,
      name: 'Walker Ranch',
      stars: 3.9,
      difficulty: 'blueBlack',
      descent: -1510,
      conditionDate: '2019-09-19 07:48:20',
      conditionStatus: 'All Clear',
    },
  ];

  fetchData.mockImplementation(() => Promise.resolve(mockTrails));

  beforeEach(() => {
    wrapper = shallow(<App getTrails={mockGetTrails} />);
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should return array of trails for based on lon and lat interpolated', async () => {
    const mockLat = 39.7392;
    const mockLon = 104.9903;
    const mockUrl = `https://www.mtbproject.com/data/get-trails?lat=${mockLat}&lon=-${mockLon}&maxDistance=10&key=200628346-0f130fc8870531d529e09b85e721317a`;

    await wrapper.instance().getLatLon(mockLat, mockLon);

    expect(trails).toEqual(mockTrails)
  });

  it('should call fetchData, filteredTrailData, and getTrails when getLatLon is called', () => {
    const mockLat = 39.7392;
    const mockLon = 104.9903;
    const mockUrl = `https://www.mtbproject.com/data/get-trails?lat=${mockLat}&lon=-${mockLon}&maxDistance=10&key=200628346-0f130fc8870531d529e09b85e721317a`;
    const filteredTrailData = jest.fn();
    const getTrails = jest.fn();

    wrapper.instance().getLatLon(mockLat, mockLon);

    expect(fetchData).toHaveBeenCalledWith(mockUrl);
    expect(filteredTrailData).toHaveBeenCalledWith(mockTrails);
    expect(getTrails).toHaveBeenCalledWith(mockTrails);
  });
});

describe('mapDispatchToProps', () => {
  it('calls dispatch when a getTrails action is called', () => {
    const mockDispatch = jest.fn();
    const mockGetTrails = jest.fn();
    const filteredTrailData = jest.fn();
    const wrapper = shallow(<App getTrails={mockGetTrails} />);
    const mockTrails = [
      {
        key: 46286,
        id: 46286,
        name: 'Betasso Preserve',
        stars: 4,
        difficulty: 'greenBlue',
        descent: -829,
        conditionDate: '2019-10-03 22:57:25',
        conditionStatus: 'All Clear',
      },
      {
        key: 601365,
        id: 601365,
        name: 'Walker Ranch',
        stars: 3.9,
        difficulty: 'blueBlack',
        descent: -1510,
        conditionDate: '2019-09-19 07:48:20',
        conditionStatus: 'All Clear',
      },
    ];
    const actionToDispatch = getTrails('GET_TRAILS', mockTrails);
    const mappedProps = mapDispatchToProps(mockDispatch);

    mappedProps.getTrails('GET_TRAILS', mockTrails);

    expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
  });
});
