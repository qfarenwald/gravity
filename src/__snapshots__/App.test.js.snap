// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match snapshot 1`] = `
<div
  className="App"
>
  <Route
    path="/"
    render={[Function]}
  />
  <Route
    exact={true}
    path="/"
    render={[Function]}
  />
  <Route
    exact={true}
    path="/trails/:location"
    render={[Function]}
  />
  <Route
    exact={true}
    path="/shuttle/:trail"
    render={[Function]}
  />
  <Route
    exact={true}
    path="/confirmation"
    render={[Function]}
  />
</div>
`;
